### PR TITLE
[usb] Add USB-to-JS adaptor and get devices via it

### PR DESCRIPTION
**Pull request by emaxx-google**
Build the foundation of our new libusb implementation: the JavaScript
interface "LibusbToJsApiAdaptor". Add its first implementation - the chrome.usb
API based one: "LibusbToChromeUsbAdaptor". Add the first method -
"listDevices". Wire it up with the C++-to-JS message channel and reroute the
C++ code to use this method.

It's preparatory work for the WebUSB support effort tracked by #429.

### CODE REVIEW

File: smart_card_connector_app/src/background.js [hunk 1]
```diff
@@ -130,9 +130,9 @@ const libusbProxyReceiver = new GSC.LibusbProxyReceiver(
 const pcscLiteReadinessTracker =
     new GSC.PcscLiteServerClientsManagement.ReadinessTracker(
         executableModule.getMessageChannel());
-const messageChannelPool = new GSC.MessageChannelPool;
+const messageChannelPool = new GSC.MessageChannelPool();
```
Comment (Fabian-Sommer) [inline smart_card_connector_app/src/background.js:133]: Should this be part of this CL? (also below)

**Review by Fabian-Sommer — APPROVED** [review]
LGTM % nits

File: third_party/libusb/webport/src/libusb-proxy-receiver.js [hunk 2]
```diff
@@ -21,6 +21,8 @@ goog.provide('GoogleSmartCard.LibusbProxyReceiver');
 
 goog.require('GoogleSmartCard.DebugDump');
 goog.require('GoogleSmartCard.Libusb.ChromeUsbBackend');
+goog.require('GoogleSmartCard.LibusbToChromeUsbAdaptor');
+goog.require('GoogleSmartCard.LibusbToJsApiAdaptor');
 goog.require('GoogleSmartCard.Logging');
 goog.require('GoogleSmartCard.RemoteCallMessage');
 goog.require('goog.Promise');
```
[the remark this answers is not in the document]
Comment (emaxx-google) [reply third_party/libusb/webport/src/libusb-proxy-receiver.js:28]: Done - Yeah, the line 35 should mention libusb_js_proxy.cc.

It not matching api_bridge.h is expected (in my previous CL I made it wrong to have the same value - it won't work as two different C++ classes can't have the same requester_name). Once I finish implementing the "adaptor", I'll delete the api_bridge completely.

**Review by emaxx-google — COMMENTED** [review]
Thanks.

**Pull request merged**